src/hotspot/share/prims/forte.cpp

```diff
@@ -578,7 +577,7 @@ void AsyncGetCallTrace(ASGCT_CallTrace *trace, jint depth, void* ucontext) {
     return;
   }
 
-  assert(JavaThread::current() == thread,
+  assert(thread == JavaThread::thread_from_jni_environment(trace->env_id),
```

Comment:
Please add  a comment before the assert:

// This is safe now as the thread has not terminated and so no VM exit check occurs.